BootloaderCommonPkg/Include/Library/SecureBootLib.h

```diff
@@ -61,6 +58,9 @@ DoHashVerify (
   @param[in]  Signature       Signature for the data buffer.
   @param[in]  PubKey          Public key data pointer.
   @param[in]  PubKeyHash      Public key hash value when ComponentType is not used.
+  @param[in]  PubKeyHashAlg   Pubkey Hash Alg .
+  @param[in]  SignKeyType     Type of Public key Sign .
+  @param[in]  SignHashAlg     Hash Alg used for signing.
   @param[out] OutHash         Calculated data hash value.
```

Comment:
Do we really support any combination of hash alg and sign type?
Currently we already defined SIG_TYPE_RSA2048_SHA256 and SIG_TYPE_RSA3072_SHA384. is it enough to only passing this kind of combined type to make API simple?

Reply:
In order to support all combinations of hash and sign types, we should keep hashalg and signtype as seperate parameters. We could remove SIG_TYPE_RSA2048_SHA256  and SIG_TYPE_RSA3072_SHA384 definitions.

Reply:
Since we are redefining the interfaces anyway, let us make it right at the beginning. 
To me, it makes sense to have separate argument for hash and signing.

Reply:
Guo, I am trying to keep secure lib generic for all components to use.
I understand we might not support all combinations in signing process.
Anyways, in secure lib and IPP crypto wrappers we need to support these parameters as separate.